### PR TITLE
add tezt.4.1.0

### PR DESCRIPTION
**Pull request by rbardou**
New release of Tezt.

### CODE REVIEW

**Comment by mseri** [comment]
A number of tests are failing. E.g.
```
# [15:49:26.916] Starting test: Process.terminate with timeout
# [15:49:26.916] [process] /home/opam/.opam/4.12/.opam-switch/build/tezt.4.1.0/_build/default/test/unix/bin_catch_sigterm/main.exe
# [15:49:26.917] process exited with code 127.
# [15:49:27.918] Will now terminate process.
# [15:49:27.918] Send SIGTERM to process.
# [15:49:27.918] Process.wait returned after 1.00177 seconds from start.
# [15:49:27.918] [error] expected Process.wait to return after about 2 seconds
# [15:49:27.918] [FAILURE] (26/26, 1 failed) Process.terminate with timeout
# [15:49:27.918] Try again with: ./main.exe --verbose --file test/unix/test_process.ml --title 'Process.terminate with timeout'
# (cd _build/default/test/js && /usr/bin/node main.bc.js)
```
or
```
# File "test/js/dune", line 3, characters 0-90:
# 3 | (rule
# 4 |  (alias runtest)
# 5 |  (package tezt)
# 6 |  (deps main.bc.js)
# 7 |  (action (run node main.bc.js)))
# (cd _build/default/test/js && /usr/bin/node main.bc.js)
# [15:47:05.226] [SUCCESS] (1/25) demo
# [15:47:05.226] Starting test: fixed seed
# [15:47:05.234] [error] expected x = 3284848139645624427, got 7949467250670036511
# [15:47:05.236] [FAILURE] (2/25, 1 failed) fixed seed
# [15:47:05.236] Try again with: /home/opam/.opam/5.1/.opam-switch/build/tezt.4.1.0/_build/default/test/js/main.bc.js --verbose --file test/common/tests.ml --title 'fixed seed'
```

**Comment by rbardou** [comment]
Another one is:
```
- File "test/cram/test-logs.t", line 1, characters 0-0:
- /usr/bin/git --no-pager diff --no-index --color=always -u _build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t _build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t.corrected
- diff --git a/_build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t b/_build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t.corrected
- index 2accab0..ec59ac7 100644
- --- a/_build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t
- +++ b/_build/.sandbox/bcf5eea2216971a9ac1c361a5bc206a2/default/test/cram/test-logs.t.corrected
- @@ -57,8 +57,8 @@ Check CLI option  `no-log-timestamp`
-    Prefixes are printed by default: true
-    Log is printed with default configuration
-    [echo#0] echo 'Log is printed following the default configuration'
- -  [echo#0] Log is printed following the default configuration
-    echo#0 exited with code 0.
- +  [echo#0] Log is printed following the default configuration
-    There is no timestamp here
-    echo 'There is no timestamp nor the name of the command here'
-    There is no timestamp nor the name of the command here
```
and
```
+ /usr/bin/sudo "apt-get" "install" "-qq" "-yy" "npm"
- E: Unable to correct problems, you have held broken packages.
[ERROR] System package install failed with exit code 100 at command:
            sudo apt-get install -qq -yy npm
[ERROR] These packages are still missing: npm
```

About the `Process.terminate with timeout` issue: the test passes on my laptop, and in the CI of Tezt. Somehow `Sys.set_signal Sys.sigterm (Signal_handle on_sigterm)` seems to not work in the opam CI. Also, this test is not critical. I think it already failed for the previous release in opam and we decided to merge anyway. I suggest we do the same. Do you think I should disable the test for the next opam release?

About the fixed seed test: the checks that `Random.int64` returns a specific value, but this is not the true goal of this test. I guess `Random.int64` changed in OCaml 5.1 or something? In any case, this failure is not worrying. I took note of the value and will try to make the test more permissive (accept both values) next time.

About the `echo` cram test: the test is known to be flaky — it happened in our CI and restarting the test fixed it. Unfortunately this is a concurrency issue that causes the output to sometimes be in a different order. I guess we could `| sort` the output in the cram test, or disable the test. Anyway, not a worrying issue either.

About the `npm` issue: I guess node.js is just not available on those architectures? (https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/204dd21cb531f052cd8d750304f63887af44eacb/variant/extras,riscv64-ocaml-4.14,tezt.4.1.0,tests and https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/204dd21cb531f052cd8d750304f63887af44eacb/variant/extras,x86_32-ocaml-4.14,tezt.4.1.0,tests)

In conclusion, I think it's ok to merge, but I opened this issue https://gitlab.com/nomadic-labs/tezt/-/issues/34 to try to fix this for the next opam release of Tezt.
